src/dist/replication/meta_server/meta_bulk_load_service.cpp

```diff
@@ -1215,39 +1213,85 @@ void bulk_load_service::on_control_bulk_load(control_bulk_load_rpc rpc)
     }
 }
 
-// ThreadPool: THREAD_POOL_META_STATE
-void bulk_load_service::create_bulk_load_root_dir(error_code &err, task_tracker &tracker)
+// ThreadPool: THREAD_POOL_META_SERVER
+void bulk_load_service::create_bulk_load_root_dir()
 {
     blob value = blob();
-    _meta_svc->get_remote_storage()->create_node(
-        _bulk_load_root,
-        LPC_META_CALLBACK,
-        [this, &err, &tracker](error_code ec) {
-            if (ERR_OK == ec || ERR_NODE_ALREADY_EXIST == ec) {
-                ddebug_f("create bulk load root({}) succeed", _bulk_load_root);
-                sync_apps_bulk_load_from_remote_stroage(err, tracker);
-            } else if (ERR_TIMEOUT == ec) {
-                dwarn_f("create bulk load root({}) failed, retry later", _bulk_load_root);
-                tasking::enqueue(
-                    LPC_META_STATE_NORMAL,
-                    _meta_svc->tracker(),
-                    std::bind(&bulk_load_service::create_bulk_load_root_dir, this, err, tracker),
-                    0,
-                    std::chrono::seconds(1));
-            } else {
-                err = ec;
-                dfatal_f(
-                    "create bulk load root({}) failed, error={}", _bulk_load_root, ec.to_string());
+    std::string path = _bulk_load_root;
+    _sync_bulk_load_storage->create_node(std::move(path), std::move(value), [this]() {
+        ddebug_f("create bulk load root({}) succeed", _bulk_load_root);
+        sync_apps_bulk_load_from_remote_storage();
+    });
+}
+
+// ThreadPool: THREAD_POOL_META_STATE
+void bulk_load_service::sync_apps_bulk_load_from_remote_storage()
+{
+    std::string path = _bulk_load_root;
+    _sync_bulk_load_storage->get_children(
+        std::move(path), [this](bool flag, const std::vector<std::string> &children) {
+            if (flag && children.size() > 0) {
+                ddebug_f("There are {} apps need to sync bulk load status", children.size());
+                for (const auto &elem : children) {
+                    uint32_t app_id = boost::lexical_cast<uint32_t>(elem);
```

Comment:
`boost::lexical_cast ` will throw exception when cast failed, are you sure the cast always success for `elem`?

Reply:
Yes, `elem` is app_id